ext/serializer.c

```diff
@@ -607,6 +607,28 @@ static zend_string *dd_get_user_agent() {
     return ZSTR_EMPTY_ALLOC();
 }
 
+static zend_string *dd_get_mapped_peer_service(zend_string *peer_service) {
+    zend_array *peer_service_mapping = get_DD_TRACE_PEER_SERVICE_MAPPING();
+    if (zend_hash_num_elements(peer_service_mapping) == 0) {
+        return NULL;
+    }
+
+    // The mappings of peer_service_mapping are strings in the format <peer_service>:<mapped_service>
+    zend_string *mapped_service = NULL;
+    zend_string *mapping;
+    ZEND_HASH_FOREACH_STR_KEY(peer_service_mapping, mapping) {
+        if (mapping && memcmp(ZSTR_VAL(mapping), ZSTR_VAL(peer_service), ZSTR_LEN(peer_service)) == 0) {
```

Comment:
I think that this would be true, while it should be false, in the following scenario:
- `peer_service = something`
- `mapping = something_and_something_else:hello`

Reply:
Indeed!